cpp/src/gandiva/precompiled/string_ops.cc

```diff
@@ -1642,6 +1642,104 @@ const char* convert_toUTF8(int64_t context, const char* value, int32_t value_len
   return value;
 }
 
+// Calculate the levenshtein distance between two string values
+FORCE_INLINE
+gdv_int32 levenshtein_utf8_utf8(int64_t context, const char* in1, int32_t in1_len,
+                                const char* in2, int32_t in2_len) {
+  if (in1_len < 0 || in2_len < 0) {
+    gdv_fn_context_set_error_msg(context, "String length must be greater than 0");
+    return 0;
+  }
+
+  // Check input size 0
+  if (in1_len == 0) {
+    return in2_len;
+  }
+  if (in2_len == 0) {
+    return in1_len;
+  }
+
+  // A and B is one copy from lengths
+  int A;
+  int B;
+
+  // 'smaller' Is a variable to validate the inversion of entries without a copy of them.
+  bool smaller = false;
```

Comment:
You can do better than this; define the following:
char *arr_larger; // points to the larger array
char *arr_smaller; // points to the smaller array

Similarly, len_larger and len_smaller

Do one if-condition like this and initialise arr_larger, arr_smaller, len_larger, len_smaller